### PR TITLE
feat: support init-templatedir

### LINKED ISSUE #32: Implement `pre-commit init-templatedir`

### PR DESCRIPTION
**Pull request by XmchxUp**
Closed #32

### CODE REVIEW

File: src/cli/mod.rs [hunk 1]
```diff
@@ -339,3 +339,17 @@ pub(crate) struct GenerateShellCompletionArgs {
     #[arg(value_enum)]
     pub shell: clap_complete::Shell,
 }
+
+#[derive(Debug, Args)]
+pub(crate) struct InitTemplateDirArgs {
+    /// The directory in which to write the hook script.
+    pub(crate) directory: String,
```
Comment (j178) [inline src/cli/mod.rs]:
```suggestion
    pub(crate) directory: PathBuf,
```

File: src/cli/mod.rs [hunk 2]
```diff
@@ -339,3 +339,17 @@ pub(crate) struct GenerateShellCompletionArgs {
...
+
+    /// Assume cloned repos should have a `pre-commit` config.
+    #[arg(long, default_value_t = false)]
```
Comment (j178) [inline src/cli/mod.rs]:
```suggestion
    #[arg(long)]
```

Comment (XmchxUp) [reply src/cli/mod.rs:349]: I think this is an option, not an argument.

File: src/cli/mod.rs [hunk 3]
```diff
@@ -339,3 +339,17 @@ pub(crate) struct GenerateShellCompletionArgs {
...
+    pub(crate) no_allow_missing_config: bool,
+
+    /// Which hook type to install.
+    #[arg(long, short = 't', default_value_t = HookType::PreCommit)]
+    pub(crate) hook_type: HookType,
```
Comment (j178) [inline src/cli/mod.rs]:
```suggestion
    #[arg(short = 't', long = "hook-type", value_name = "HOOK_TYPE", value_enum)]
    pub(crate) hook_types: Vec<HookType>,
```

**Comment by j178** [comment]
Thanks! Could you add a test case too?

**Comment by XmchxUp** [comment]
> Thanks! Could you add a test case too?

Thanks for the review, will add it later.

**Comment by XmchxUp** [comment]
@j178 

Should `init-templatedir`, which is based on `install`, use a mock library to simulate the behavior of install for testing, or is there another method?

**Comment by j178** [comment]
We don’t need to use mocks in integration tests. You can refer to other tests in the `tests` folder for guidance.

**Comment by j178** [comment]
I made some tweaks and added a test, thanks for your contribution!

**Pull request merged**